src/Utilities/FlowAnalysis/FlowAnalysis/Framework/DataFlow/AbstractDataFlowAnalysisContext.cs

```diff
@@ -40,6 +40,7 @@ public abstract class AbstractDataFlowAnalysisContext<TAnalysisData, TAnalysisCo
             InterproceduralAnalysisData<TAnalysisData, TAnalysisContext, TAbstractAnalysisValue> interproceduralAnalysisDataOpt,
             InterproceduralAnalysisPredicate interproceduralAnalysisPredicateOpt)
         {
+            Debug.Assert(valueDomain != null, "valueDomain must not be null for use in ComputeHashCodeParts");
```

Comment:
Nit: not sure the message is useful here. ValueDomain is one of the core pieces of merging DFA values, so the assert seems self contained.

Reply:
Ok. I figured you could figure out which parameter w/o attaching a debugger by giving a message.